File: cpp/Player.cpp
// Honor Pledge:
//
// I pledge that I have neither given nor 
// received any help on this assignment.
//
// banspate

#include <iostream>
#include <string>
#include <stdlib.h>
#include <ctime>
#include "Player.h"

//default constructor
Player::Player(){
	
}

//destructor
Player::~Player(){
	
	//delete the block of array stored in heap to avoid memory leaks
	delete[] array;
	
}

Player::Player(std::string name, int takeShot){
	
	//instance of array
	array = new char[5];
	
	//allocate value for each array element
	array [0] = 'H';
	array [1] = 'O';
	array [2] = 'R';
	array [3] = 'S';
	array [4] = 'E';
	
	i = 0;

	//setting the name passed as parameter in class player
	this -> name = name;
	
}

//getter method used to access player name
std::string Player::getPlayerName(){
	
	//getting the name from the player
	return name;
	
}

//method that helps player make hit/ take shot
int Player::makeHit(){
	
	//generates a random number either 2 or 1
	int takeShot = rand () %2 + 1;
	
	return takeShot;
	
}

// method that helps keep track of characters that each player added from word HORSE 
char Player::playerScore(){
	
	std::cout << ( array[i] ) << std::endl;
			
	i++;
		
	return 0;	
	
}

// if the i reaches the value 5 its outside the array bound hence it returns true
//which inturn means player looses game
bool Player::loose(){
			
	if( i == 5 ){
			
		return true;
		
	}else{
			
		return false;
			
	}
}
		

File: Java/Player.java
// Honor Pledge:
//
// I pledge that I have neither given nor 
// received any help on this assignment.
//
// banspate

import java.util.Random;


// Class Declaration

public class Player{
	
	// Instance Variables
	String name;
	int takeShot;
	char [] array = {'H','O','R','S','E'};
	
	
	int i = 0;
	
	// overloaded constructor Declaration of Class
	public Player(String name, int takeShot){   
	
		this.name = name;
		
	}
	// method 1
	public String getName(){
		
		return name;
	}
	// method 2
	public int makeHit(){
		//generate random number either 1 or 2
		Random rand = new Random();
		this.takeShot = rand.nextInt(2) + 1;
		
		return this.takeShot;
		
	}

	//method 3
	public String toString(){
		// print the player name
		return( this.getName() + ":" );
	}
	
	//method 4
	public char playerScore(){
		//print the character from HORSE
		System.out.println( array[i] );
		
		i++;
		
		return 0;
		
	}
	//method 5
	
	public boolean loose(){
		
		//decide if player looses or wins
		//as soon as i goes to 5th index it breaks
		if( i == 5 ){
			
			return true;
		
		}else{
			
			return false;
			
		}
		
	}

}